### DISCUSSION TITLE
Add vyper support

### DISCUSSION THREAD

**Discussion by roninjin10** [evmts/tevm-monorepo#1264] [Ideas]
Add support for viper. Solc works with viper so this should be a smallish lift.

What I think is needed:
[] Add viper extension support to the @evmts/ts-plugin
[] Add viper extension support to @evmts/bundler
[] Update solc compiler in @evmts/bundler if necessary

**Comment by roninjin10** [comment]
this is harder than I expected

**Comment by roninjin10** [comment]
We are so back

**Comment by roninjin10** [comment]
z80 🐍 💯, [May 16, 2024 at 11:26:11 AM]:
is it at all like bun custom loaders? I made one of those for vyper, I would like to contribute a vyper one to tevm

looks like this in bun
![telegram-cloud-photo-size-1-5057962262291131479-y](https://github.com/evmts/tevm-monorepo/assets/35039927/511eb740-5ea8-4b33-b889-96a4d37a7ae8)

![telegram-cloud-photo-size-1-5057962262291131479-y](https://github.com/evmts/tevm-monorepo/assets/35039927/511eb740-5ea8-4b33-b889-96a4d37a7ae8)

**Comment by roninjin10** [comment]
That would be dope!

The tevm bundler supports following bundlers:

- bun
- esbuild
- webpack / rspack
- vite / rollup

The core bundler @tevm/bundler is not tightly coupled to any bundler and then reused to implement any specific build tool

**Comment by roninjin10** [comment]
Either way the MVP of vyper support I bet we could get done in an hour or 2 of work given what you built already

**Comment by roninjin10** [comment]
> On whether deployedBytecode exists

both, we call it bytecode_runtime

**Comment by z80dev** [comment]
gonna tackle this tonight

**Comment by roninjin10** [comment]
It's not released yet but we have vyper support now so gonna close this via our compilers package. 

tevm/compilers now just wraps foundry/compilers rust crate. Since foundry supports vyper we get it for free
